libs/src/api/entityManager.ts

```diff
@@ -0,0 +1,327 @@
+const MAX_PLAYLIST_ID = 2147483647
+
+/*
+  API surface for updated data contract interactions.
+  Provides simplified entity management in a generic fashion
+  Handles metadata + file upload etc. for entities such as Playlist/Track/User
+*/
+export class EntityManager extends Base {
+  constructor(...args: BaseConstructorArgs) {
+    super(...args)
+    this.getValidPlaylistId = this.getValidPlaylistId.bind(this)
+    this.createPlaylist = this.createPlaylist.bind(this)
+    this.deletePlaylist = this.deletePlaylist.bind(this)
+  }
+
+  /**
+   * Generate random integer between two known values
+   */
+  getRandomInt(min: number, max: number): number {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min) + min)
+  }
+
+  /**
+   * Calculate an unoccupied playlist ID
+   * Maximum value is postgres integer max (2147483647)
+   * Minimum value is artificially set to 400000
+   */
+  async getValidPlaylistId(): Promise<number> {
+    let playlistId: number = this.getRandomInt(MIN_PLAYLIST_ID, MAX_PLAYLIST_ID)
+    let validIdFound: boolean = false
+    while (!validIdFound) {
+      const resp: any = await this.discoveryProvider.getPlaylists(1, 0, [
+        playlistId
+      ])
+      if (resp.length !== 0) {
+        playlistId = this.getRandomInt(MIN_PLAYLIST_ID, MAX_PLAYLIST_ID)
+      } else {
+        validIdFound = true
+      }
+    }
+    return playlistId
+  }
+
+  /**
+   * Playlist default response values
+   */
+  getDefaultPlaylistReponseValues(): PlaylistOperationResponse {
+    return {
+      blockHash: null,
+      blockNumber: null,
+      playlistId: null,
+      error: null
+    }
+  }
+
+  /**
+   * Create a playlist using updated data contracts flow
+   */
+  async createPlaylist({
+    playlistName,
+    trackIds,
+    description,
+    isAlbum,
+    isPrivate,
+    coverArt,
+    logger = console
+  }: {
+    playlistName: string
+    trackIds: number[]
+    description: string
+    isAlbum: boolean
+    isPrivate: boolean
+    coverArt: string
+    logger: Console
+  }): Promise<PlaylistOperationResponse> {
+    const responseValues: PlaylistOperationResponse =
+      this.getDefaultPlaylistReponseValues()
+    try {
+      const currentUserId: string | null =
+        this.userStateManager.getCurrentUserId()
+      if (!currentUserId) {
+        return {
+          blockHash: null,
+          blockNumber: null,
+          playlistId: null,
+          error: 'Missing current user ID'
+        }
+      }
+      const userId: number = parseInt(currentUserId)
+      const createAction = Action.CREATE
+      const entityType = EntityType.PLAYLIST
+      const entityId = await this.getValidPlaylistId()
+      this.REQUIRES(Services.CREATOR_NODE)
+      const updatedPlaylistImage = await this.creatorNode.uploadImage(
+        coverArt,
+        true // square
+      )
+      const dirCID = updatedPlaylistImage.dirCID
+      const metadata = {
+        createAction,
```

Comment:
do we want to keep this create action in here - if so, should probably be camelcase to match the format of the rest of the fields